### ISSUE TITLE
Broken docgen target: //src/main/java/com/google/devtools/build/lib:gen_command-line-reference

### ISSUE THREAD

**Issue by jin** [bazelbuild/bazel#4730]
```
$ bazel build //site --verbose_failures
INFO: Analysed target //site:site (0 packages loaded).
INFO: Found 1 target...
ERROR: /usr/local/google/home/jingwen/bazels/bazel_github/src/main/java/com/google/devtools/build/lib/BUILD:1343:1: Executing genrule //src/main/java/com/google/devtools/build/lib:gen_command-line-reference failed (Exit 37): bash failed: error executing command
  (cd /usr/local/google/home/jingwen/.cache/bazel/_bazel_jingwen/e8b94c2d43a1b52123cfa888c6b59ca8/execroot/io_bazel && \
  exec env - \
    PATH= (redacted) \
  /bin/bash -c 'source external/bazel_tools/tools/genrule/genrule-setup.sh; cat site/command-line-reference-prefix.html > bazel-out/k8-fastbuild/genfiles/src/main/java/com/google/devtools/build/lib/command-line-reference.html && TMP=`mktemp -d /tmp/tmp.XXXXXXXXXX` && mkdir ${TMP}/_embedded_binaries/ && bazel-out/host/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer --jvm_flag=-Dio.bazel.EnableJni=0 --batch --install_base=${TMP} --output_base=${TMP}/output/ help everything-as-html >> bazel-out/k8-fastbuild/genfiles/src/main/java/com/google/devtools/build/lib/command-line-reference.html 2>/dev/null && cat site/command-line-reference-suffix.html >> bazel-out/k8-fastbuild/genfiles/src/main/java/com/google/devtools/build/lib/command-line-reference.html')

Use --sandbox_debug to see verbose messages from the sandbox
Target //site:site failed to build
INFO: Elapsed time: 0.962s, Critical Path: 0.63s
FAILED: Build did NOT complete successfully
```

**Comment by jin** [comment]
cc @katre

**Comment by jin** [comment]
Found the commit that introduced the regression: c2499c406d0727853ea0ddc5372a50be95de5515

6e0933886d3c6b7f68075da4bdb08500ce2b6f86 depends on it.

cc @aehlig

**Comment by aehlig** [comment]
> cc @aehlig, do you know what about your change might have caused this?

Yes. The problem is that here we have a direct, undocumented, call to the bazel server, which was considered an internal interface. Well, it is an internal call, so it probably is OK. Still annoying that none of our usual presubmit tests found it.

Anyway, this is a fix which I plan to submit later today

```From 331cb4e3dc707945d41307ebd883fb5fc4d909fb Mon Sep 17 00:00:00 2001
From: Klaus Aehlig <aehlig@linta.de>
Date: Wed, 28 Feb 2018 08:43:50 +0100
Subject: [PATCH] Fix additional direct use of the Bazel server

...by adding the new internal startup option --output_user_root.

Fixes #4730.
---
 src/main/java/com/google/devtools/build/lib/BUILD | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main/java/com/google/devtools/build/lib/BUILD b/src/main/java/com/google/devtools/build/lib/BUILD
index 553c8f3907..433aacf7d6 100644
--- a/src/main/java/com/google/devtools/build/lib/BUILD
+++ b/src/main/java/com/google/devtools/build/lib/BUILD
@@ -1353,7 +1353,7 @@ genrule(
         "mkdir $${TMP}/_embedded_binaries/ && " +
         "$(location :bazel/BazelServer) " +
         "--jvm_flag=-Dio.bazel.EnableJni=0 --batch " +
-        "--install_base=$${TMP} --output_base=$${TMP}/output/ " +
+        "--install_base=$${TMP} --output_base=$${TMP}/output/ --output_user_root=$${TMP} " +
         "help everything-as-html >> $@ 2>/dev/null && " +
         "cat $(location //site:command-line-reference-suffix.html) >> $@"
     ),
-- 
2.16.2
```
